### ISSUE TITLE
Exporting trees from Hyphy Vision with node data?

### ISSUE THREAD

**Issue by podunn58** [veg/hyphy#704]
Hi 
I am using aBSREL on the terminal and viewing my JSON file at Hyphy Vision.  The results are complex and crowded, so I want to clean it up in a tree drawing program.  Unfortunately, I do not see an easy way to export the node values for selection (w) with the Newick tree.  Does someone know of an easy way to do this or, alternatively, add the w values to the tree afterwards (maybe in R)?  Sorry, I know this is probably simple, but I'm a beginner at this, and I can't seem to find an answer.
Thanks, 
Peter

**Comment by sjspielman** [comment]
Dear Peter,

I actually have a Python package under development to do just this, and similar tasks, at github.com/sjspielman/phyphy. I expect to write full documentation and release it within the next 2-3 weeks, so please feel free to follow the repo for updates. If you would like you can contact me directly with your JSON and goals I can help you get done what you need pre-release.

Best,
Stephanie 

Best, 
Stephanie

**Comment by stevenweaver** [comment]
Hi Peter, 

As it stands, we do not have a method to export Newick trees with omega values. The pages are quite new so requests are more than welcome. I agree that having different newick formatting options would be useful.

In the meantime, if you do want to parse information in R, you should be able to parse the JSON directly. 

For example, within the [aBSREL example file](http://vision.hyphy.org/methods/absrel/data/ABSREL.json), you will see a field called `branch attributes`, with entries like the following :

```
     "0564_11":{
       "original name":"0564_11",
       "Nucleotide GTR":0.004348809000618374,
       "Baseline MG94xREV":0.004292442332284749,
       "Baseline MG94xREV omega ratio":0.4118756279759579,
       "Rate classes":2,
       "Full adaptive model":0.005272679063666866,
       "Rate Distributions":[
        [0, 0.9860672522817131],
        [42.79984883126327, 0.0139327477182869]
        ],
       "LRT":3.960489769506239,
       "Uncorrected P-value":0.05072665887336775,
       "Corrected P-value":1
      },
```

The field `Rate Distributions` correspond to omega values and their respective proportions in order. 
Notice how 0 and .986 and 42.8 and 0.01 correspond to the view. 


<img width="1088" alt="screen shot 2017-11-24 at 5 57 54 pm" src="https://user-images.githubusercontent.com/535261/33225196-0ab2aaec-d141-11e7-89de-c302ce052a7d.png">

Hopefully this helps a bit if you are looking to do something now. I can put together a quick script to format a new Newick string from this information if you think that could still be useful after this comment.

Best,
Steven

**Comment by podunn58** [comment]
Hi Steven and Stephanie
Thanks for your quick help.  It sounds like Stephanie is producing a script to do what Steven is suggesting.  In any case, I think it would be very helpful to have the ability to export omega values with the tree.  
Best,
Peter

**Comment by spond** [comment]
Dear @podunn58,

Please note that the Newick format for trees does not formally support general node annotation (other than branch lengths) Various programs, such as FigTree, have developed custom annotations (e.g. using the `[]` comments). Do you have an idea which of the non-standard extensions you might need?

Also, I would be curious to hear what other visualizations we might include in the standard HyPhy vision pages to make it more generally useful.

Best,
Sergei

**Comment by podunn58** [comment]
Hi Sergei et al.
Thanks for the quick responses, and clarification about Newick format.  I ended up importing the tree into R and adding omega values (imported from a separate file) to the node labels.  Then I saved it as nexus, which partly worked, but there are some things I need to fix. I have used FigTree, too, but I have not figured out how to do the annotation the way I want.  Ultimately, what I would like to do is color the branches in the tree based on the omega values (as shown in HyPhy Vision). Hyphy vision is probably fine for most people, but my file has 300+ sequences and short branches, so it is hard to see on HyPhy Vision (even in the Radial view). It looks okay in Figtree, so it would be nice to have the ability to manipulate the tree more in another program.  Adding the ability to rescale branch lengths might help in HyPhy Vision. I don't know how much trouble this would be, but the ability to save the tree and omega values in a Nexus file would also be a nice Christmas present.
Thanks,
Peter
